bolt-simple-taproot.md

```diff
@@ -1163,137 +1231,117 @@ A new TLV stream is added to the `revoke_and_ack` message:
+   1. type: 24 (`current_commit_nonce`)
+   2. data:
+       * [`66*byte`:`public_nonce`]
 
-Similar to the `next_per_commitment_point`, by sending the `next_local_nonce`
-value in this message, we ensure that the remote party has our public nonce,
-which is required to generate a new commitment signature.
+Similar to the `next_per_commitment_point`, by sending the `next_local_nonces`
+value in this message, we ensure that the remote party has our public nonces,
+which are required to generate new commitment signatures.
+
+`current_commit_nonce` is used to re-send signatures for the * current * 
+commitment transaction.
 
 ##### Requirements
 
 The sender:
 
-- MUST set `next_local_nonce` to a fresh, unique `musig2` nonce as specified by
-  `bip-musig2`
-- For taproot channels, SHOULD also populate the `local_nonces` field:
-  - MUST include at least one entry with an empty hash (32 zero bytes) as the key,
-    containing the primary commitment nonce
-  - The value for the empty hash key MUST match the value in `next_local_nonce`
-  - MAY include additional entries for in-progress splice transactions
-  - MUST sort entries by TXID in lexicographical order when encoding
+  - MUST include one entry for each active commitment transaction,
+    indexed by the commitment transaction's funding txid.
+  - if it has sent commitment_signed for an interactive transaction 
+    construction but it has not received tx_signatures:
+      - MUST include `current_commit_nonce` for the commitment transaction that is being built.
 
 The recipient:
 
-- MUST fail the channel if `next_local_nonce` is absent, or cannot be parsed as
-  two compressed secp256k1 points.
-- When `local_nonces` field is present:
-  - MUST prioritize `local_nonces` over `next_local_nonce` for obtaining the
-    commitment nonce
-  - MUST use the nonce associated with the empty hash key (32 zero bytes) as the
-    primary commitment nonce
-  - MAY store additional nonces for splice coordination
-- For taproot channels, if neither `next_local_nonce` nor `local_nonces` contains
-  a valid nonce, MUST fail the channel
+- MUST fail the channel if `next_local_nonces` is absent, or cannot be parsed.
+- MUST fail the channel if an active commitment funding txid is missing in
+  `next_local_nonces`.
+- if `channel_reestablish` includes a `next_funding_txid` which matches the
+  latest interactive funding transaction or the current channel funding transaction
+  - if it must re-transmit its `commitment_signed`
+    - if `current_commit_nonce` is missing: MUST fail the channel
 
-A node:
-
-  - If ALL of the following conditions apply:
-  
-    - It has previously sent a `commitment_signed` message
-  
-    - It never processed the corresponding `revoke_and_ack` message
-  
-    - It decides to retransmit the exact `update_` messages from the last sent
-      `commitment_signed`
-  
-  - THEN it must regenerate the partial signature using the newly received
```

Comment:
This is super important. Otherwise retransmission doesn't work (invalid sig), as the other party will generate fresh nonces on startup.